pkg/types/gcp/platform.go

```diff
@@ -59,4 +59,28 @@ type Platform struct {
 	// such as the current env OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE
 	// +optional
 	Licenses []string `json:"licenses,omitempty"`
+
+	// PrivateDNSZone Technology Preview.
+	// When the PrivateDNSZone is empty, the DNS zones will be created and managed in the Service
+	// Project (ProjectID). When PrivateDNSZone is provided, the DNS zone will be created and managed
+	// in the requested Project through PrivateZoneProject.
+	// +optional
+	PrivateDNSZone string `json:"privateDNSZone,omitempty"`
+
+	// PrivateZoneProject Technology Preview
+	// When the PrivateDNSZone is provided, the zone will be created in this project.
+	// +optional
+	PrivateZoneProject string `json:"privateZoneProject,omitempty"`
```

Comment:
If we are going to use multiple fields, we might want to consider joining them in a struct
```
type DNSZone struct {
    ID string `json:"id,omitempty"`
    ProjectID string `json:"project,omitempty"`
}
```

and
```
PublicDNSZone DNSZone `json:"publicDNSZone,omitempty"`
PrivateDNSZone DNSZone `json:"privateDNSZone,omitempty"`
```